internal/blobinfocache/types.go

```diff
@@ -25,6 +25,16 @@ type BlobInfoCache2 interface {
 	// Close destroys state created by Open().
 	Close()
 
+	// UncompressedDigest returns an uncompressed digest corresponding to anyDigest.
+	// Returns "" if the uncompressed digest is unknown.
+	// FIXME: Does this need to record TOC/compression type?
+	UncompressedDigestForTOC(tocDigest digest.Digest) digest.Digest
```

Comment:
The TOC digest is the checksum of the uncompressed JSON document, so I think the compression should not matter in this case

Reply:
I agree we probably don’t need that right now (with `GetTOCDigest` refusing to work on manifests which contain multiple TOC digest annotations, and presumably with the zstd / estargz code being unable to decompress the other one).

This comment is a looking a bit more into the future, for lookups in the other direction, where we will want to look up (UncompressedDigest → (compressed digest, TOC digest, algorithm)) and match that against “the user wants the destination to contain zstd:chunked” (i.e. reject estargz matches).

Reply:
>  for lookups in the other direction, 

That will be done in a separate data structure (an extension of `RecordDigestCompressorName`: We need the full set of annotations for reuse of a TOC-compressed blob, so this simple mapping is not sufficient anyway. And the other structure does record the algorithm.